### PR TITLE
Make it easy to know where to modify when updating Sidebar item

### PR DESCRIPTION
**Pull request by sangwoo108**
There are several places to visit when adding a new Sidebar built-in item. This patch makes it easy to know where to modify when updating Sidebar item by checking preconditions at compile time. Also this PR adds kPlaylist type.

Resolves https://github.com/brave/brave-browser/issues/25306

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/sidebar/sidebar_service.cc [hunk 1]
```diff
@@ -48,35 +48,50 @@ SidebarItem GetBuiltInItemForType(SidebarItem::BuiltInItemType type) {
...
-      return SidebarItem::Create(GURL("chrome://history/"),
-                                 brave_l10n::GetLocalizedResourceUTF16String(
-                                     IDS_SIDEBAR_HISTORY_ITEM_TITLE),
-                                 SidebarItem::Type::kTypeBuiltIn,
-                                 SidebarItem::BuiltInItemType::kHistory, true);
-    default:
+          SidebarItem::BuiltInItemType::kReadingList,
+          /* open_in_panel = */ true);
+    case SidebarItem::BuiltInItemType::kHistory: {
+      // TODO(sko) When should we show history item?
```
Comment (simonhong) [inline components/sidebar/sidebar_service.cc:75]: Maybe we'll have another history page that fits for sidebar in the future.

File: components/sidebar/sidebar_service.cc [hunk 2]
```diff
@@ -48,35 +48,50 @@ SidebarItem GetBuiltInItemForType(SidebarItem::BuiltInItemType type) {
...
+                                       IDS_SIDEBAR_HISTORY_ITEM_TITLE),
+                                   SidebarItem::Type::kTypeBuiltIn,
+                                   SidebarItem::BuiltInItemType::kHistory,
+                                   /* open_in_panel = */ true);
+      } else {
+        return SidebarItem();
+      }
+    }
+    case SidebarItem::BuiltInItemType::kPlaylist: {
+      NOTIMPLEMENTED();
```
Comment (simonhong) [inline components/sidebar/sidebar_service.cc:89]: nit: How about deleting this? This will make a lot of log logs in console.

**Review by simonhong — APPROVED** [review]
Nice cleanup! 👍🏼

File: components/sidebar/sidebar_service.cc (hunk 1 above)
Comment (sangwoo108) [reply components/sidebar/sidebar_service.cc:75]: Yeah, It seems that upstream folks are implementing this too.

File: components/sidebar/sidebar_service.cc (hunk 2 above)
Comment (sangwoo108) [reply components/sidebar/sidebar_service.cc:89]: Good call. Done.

**Pull request merged**